core/src/actors/sessions_manager.rs

```diff
@@ -116,6 +114,17 @@ impl SessionsManager {
             .map(actix::fut::ok)
             .unwrap_or_else(|| actix::fut::err(()))
     }
+
+    /// Method to process the configuration received from the config manager
+    fn process_config(config: &Config, act: &mut Self, _ctx: &mut <Self as Actor>::Context) {
```

Comment:
the signature is a bit strange, why not
```suggestion
    fn process_config(&mut self, config: &Config, _ctx: &mut <Self as Actor>::Context) {
```
and `process_config` impl'd for several managers, but the parameters are optional sometimes, it's not a trait method, why keep the signature unchanged as unneeded `_ctx` in this case?

Reply:
I agree that the signature of the `process_config` methods looks strange, I will change it to your suggestion, thanks!

The signature always has those three parameters (even if unneeded) because those `process_config` functions are passed as parameters to the `send_get_config_request` function:
```
/// Method to send a GetConfig message to the ConfigManager
pub fn send_get_config_request<T, U: 'static>(act: &mut T, ctx: &mut T::Context, process_config: U)
where
    T: Actor,
    T::Context: AsyncContext<T>,
    U: FnOnce(&Config, &mut T, &mut T::Context)
```
Some `process_config` functions do need the context to perform their functionality (for instance, the one in the `ConnectionsManager`), so this was the best I could think of so that all `process_config` functions got what they needed.

I'm open to suggestions/improvements if this looks weird!